src/backend/distributed/deparser/deparse_table_stmts.c

```diff
@@ -273,6 +302,85 @@ AppendAlterTableCmdAddConstraint(StringInfo buf, Constraint *constraint,
 			appendStringInfo(buf, " NO INHERIT");
 		}
 	}
+	else if (constraint->contype == CONSTR_FOREIGN)
+	{
+		appendStringInfoString(buf, " FOREIGN KEY");
+
+		AppendColumnNameList(buf, constraint->fk_attrs);
+
+		appendStringInfoString(buf, " REFERENCES");
+
+		appendStringInfo(buf, " %s", quote_identifier(constraint->pktable->relname));
+
+		if (list_length(constraint->pk_attrs) > 0)
+		{
+			AppendColumnNameList(buf, constraint->pk_attrs);
+		}
+
+		/*
+		 * Append supported options if provided.
+		 * Note: When a constraint name is provided by the client therefore we do not deparse the statement in preprocessing,
+		 * the unsupported statements are caught in postprocessing in ErrorIfUnsupportedForeignConstraintExists function.
+		 * Hence the error messages for the unsupported alter table statements are unexpectedly different for the
+		 * ALTER TABLE ... ADD FOREIGN KEY and ALTER TABLE ...  ADD CONSTRAINT <conname> FOREIGN KEY
+		 */
+
+		/* FKCONSTR_MATCH_SIMPLE is default. Append matchtype if not default */
```

Comment:
Can we also have `FKCONSTR_MATCH_PARTIAL`?

Reply:
Apparently its not yet implemented

Reply:
We can ignore this for now, since we error out in plain PG, it's not a valid command:
`ERROR:  MATCH PARTIAL not yet implemented`

Reply:
FKCONSTR_MATCH_PARTIAL is not covered in  sql/multi_foreign_key.sql. Do we know if we should error out or support it?